services/src/main/java/com/linkedin/kmf/consumer/NewConsumer.java

```diff
@@ -30,8 +30,12 @@ public NewConsumer(String topic, Properties consumerProperties) {
 
   @Override
   public BaseConsumerRecord receive() {
-    if (_recordIter == null || !_recordIter.hasNext())
-      _recordIter = _consumer.poll(Long.MAX_VALUE).iterator();
+    if (_recordIter == null || !_recordIter.hasNext()) {
```

Comment:
This is one example where requiring `receive()` to be non-blocking makes implementation more complex.